### PR TITLE
[ENH] Dockerize tests

### PR DESCRIPTION
**Pull request by RISHIKESHAVAN**
#### What does this implement/fix? Explain your changes.
Modified the steps in `Makefile` to run docker tests. 
Added relevant documentation to run the docker tests.

#### What should a reviewer concentrate their feedback on?
Documentation.

##### For all contributions
- [x] The PR title starts with either [ENH], [DOC] or [BUG] indicating whether the PR topic is related to enhancement, documentation or bug

### CODE REVIEW

**Comment by RISHIKESHAVAN** [comment]
@lmmentel Can you please review this PR?

File: Makefile [hunk 1]
```diff
@@ -62,8 +62,16 @@ nb: clean
 	python3 -m venv .venv
 	. .venv/bin/activate && python -m pip install .[all_extras,binder] && ./build_tools/run_examples.sh
 
-dockerbuild:
-	DOCKER_BUILDKIT=1 docker build -t sktime -f build_tools/docker/Dockerfile .
+dockertest:
+
+ifeq ($(pyversion),3.7)
```
Comment (lmmentel) [inline Makefile]: How about replacing the if block with a variable passed to the command, sth like 

```
docker build -t sktime -f build_tools/docker/$(PYTHON_VERSION).dockerfile .
```

where `PYTHON_VERSION` would be `py39`

File: docs/source/developer_guide/testing_framework.rst [hunk 2]
```diff
@@ -302,3 +302,12 @@ If a module for a new estimator type is added, multiple things need to be create
   This can be modelled off ``test_all_estimators`` or ``test_all_forecasters``.
 * and, a collection of tests for interface compliance with the base class of the estimator type.
   The tests should cover positive cases, as well as testing raising of informative error message in negative cases.
+
+Local Testing
+~~~~~~~~~~~~~~
+
+Execution of tests in local has been containerized with the help of ``docker``. In order to execute the tests, you need to have ``docker`` installed in your system. If you need to install ``docker``, please follow the instructions `here <https://docs.docker.com/desktop/>`.
```
Comment (lmmentel) [inline docs/source/developer_guide/testing_framework.rst]: the last link is missing `_` at the end to be a proper link

File: docs/source/developer_guide/testing_framework.rst [hunk 3]
```diff
@@ -302,3 +302,12 @@ If a module for a new estimator type is added, multiple things need to be create
...
+
+The command to execute the tests is: ``make dockertest pyversion=<python version>``. You can specify the Python version to be used for the tests using the ``pyversion`` argument. For example, to execute the tests in the Python version ``3.7``, use ``make dockertest pyversion=3.7``. The supported versions are ``3.7``, ``3.8`` and ``3.9`` (``3.10`` is under development). If the ``pyversion`` argument is not passed, version ``3.8`` will be used by default.
+
+Once the command is executed, ``docker`` will then create a container using the python base image of the specified version and execute the tests within it.
```
Comment (lmmentel) [inline docs/source/developer_guide/testing_framework.rst:326]: This is not strictly true since the tests need to be started by a separate command to be run.

**Review by lmmentel — CHANGES_REQUESTED** [review]

File: docs/source/developer_guide/testing_framework.rst (hunk 3 above)
Comment (RISHIKESHAVAN) [reply docs/source/developer_guide/testing_framework.rst:326]: I believe you are talking about the command `docker run -it --name sktime sktime bash -c "make test"` 
This will automatically be called as part of the `dockertest` step in the `Makefile` and the user doesn't need to run any additional command for this.

**Comment by RISHIKESHAVAN** [comment]
@lmmentel  Have updated the changes.

**Review by lmmentel — APPROVED** [review]

**Comment by lmmentel** [comment]
Excellent, thank you!
